### ISSUE TITLE
feat: Support AWS 1.4

### ISSUE THREAD

**Issue by owenrumney** [aquasecurity/trivy#2734]
Tracked in defsec under https://github.com/aquasecurity/defsec/issues/858

**Comment by knqyf263** [comment]
@owenrumney Do you think it will come in v0.32.0?

**Comment by owenrumney** [comment]
Should do - there isn't too much to do, but I'm out for 2 weeks of it and @liamg  is out for 1 week. We should make it though

**Comment by knqyf263** [comment]
Noted. I set it for v0.32.0. If we don't make it, we can put it off to v0.33.0.

**Comment by knqyf263** [comment]
I'm reviewing [the v0.32.0 milestone](https://github.com/aquasecurity/trivy/milestone/9).
@owenrumney Is this going to be included in v0.32.0?

**Comment by owenrumney** [comment]
@knqyf263 - there is a question around how people switch between frameworks that needs to be resolved; I think @liamg mentioned there are a number of 1.2 and 1.4 checks that contradict and we need to resolve that.

defsec has a scanner option of `--framework` ready for this, but it needs thought.

TLDR; I think we should wait till next v0.33.0 rather than put it out with potentially contradictory results

**Comment by knqyf263** [comment]
Got it. Thanks for explaining! I moved it to v0.33.0.

**Comment by owenrumney** [comment]
> doesn't the fact that most of these rules are already merged into defsec mean that they'll end up in this release anyway?

they're there - but they won't collide at the moment. the mechanism to specify 1.4 explicitly is the issue

**Comment by itaysk** [comment]
can you please clarify - if someone does a `trivy aws` scan right now (with the CIS 1.4 checks already merged but no way to select CIS version), which check are being used?

**Comment by liamg** [comment]
@itaysk Right now we apply the default rules and also CIS AWS 1.2 rules (some of which may overlap)

**Comment by liamg** [comment]
We probably shouldn't apply 1.2 AND 1.4 at the same time as some of them collide with each other.

**Comment by itaysk** [comment]
Thanks for explaining. So why do we say that it is pushed to next release? Isn't it out already?
